### ISSUE TITLE
thinking_budget enforcement is invisible on the wire: finish_reason=stop and no reasoning-token count in usage

### ISSUE THREAD

**Issue by ElvezPelvez** [jundot/omlx#3687]
When `thinking_budget` is enforced, nothing on the wire says so. The reply comes back with `finish_reason=stop` and a `usage` block that carries no reasoning-token count, so a caller cannot tell a reply the model finished on its own from one whose thought was severed mid-sentence by the budget — and the two are not equally usable.

Measured on 0.6.4 (Homebrew, macOS, M1 Max 64 GB), `Qwen3.8-27B-oQ4e-fp16-mtp`, `temperature=0`, `max_tokens=8192`, one resident model, same prompt (~5 560 prompt tokens, German, asks for a single JSON object):

| `thinking_budget` | `reasoning_content` | visible `content` | `completion_tokens` | usable |
| --- | --- | --- | --- | --- |
| 256 | 1 113 ch (severed) | 1 478 ch, parses | 594 | yes, 3/3 |
| **1024** | 4 206 ch (severed) | **256 ch, cut mid-word** | **1 083** | **no, 3/3** |
| 2048 | 8 435 ch (severed) | 781 ch, parses | 2 222 | yes, 2/2 |

At `thinking_budget=1024` the reply ends mid-string:

```
{"effects": [{"feature": "jeder Batteriekomponentenhalter ist sandwichartig zwischen der ersten Batteriegehäusekomponente und der zweiten Batteriegehäusekomponente angeordnet", "effect": "Die erste und die zweite Batteriegehäusekomponente bilden zwei gegen
```

and `reasoning_content` ends mid-word too (`... ein unabhängiger Anspruch (Batteriegeh`). 1 024 reasoning tokens + 59 visible = the reported 1 083, against a `max_tokens` of 8 192 that was nowhere near reached.

**This is not a bug report against the budget processor.** `ThinkingBudgetProcessor` does exactly what it documents — masks logits to force `\n</think>\n\n`, then becomes a no-op — and `finish_reason=stop` is literally accurate: the model sampled a token in `stop_token_ids` (`omlx/scheduler.py:8809`). Dropped out of an unfinished sentence, the model emits its end-of-turn token 59 tokens into the answer. The 256- and 2048-token budgets prove the forced close is not itself harmful. Reproduced byte-identically with `mtp_enabled: false`, so speculative decoding is not involved.

The gap is the missing signal. A caller that wants to refuse a severed reply has nothing honest to test:

- `finish_reason` is `stop`, truthfully.
- `usage` has `prompt_tokens`, `completion_tokens`, `total_tokens`, `input_tokens`, `output_tokens`, `prompt_tokens_details.cached_tokens`, `total_time` — **no reasoning-token count**, so "the budget was hit" is not derivable.
- Arithmetic on `completion_tokens` vs the requested budget is a heuristic that fires on legitimately short answers, which is exactly what a caller must not ship.

**Asking for either of:**

1. `usage.completion_tokens_details.reasoning_tokens` (the shape the OpenAI-compatible surface already uses elsewhere), which makes "reasoning == budget" a fact the caller can read; or
2. an explicit marker that the processor forced the close on this request — a boolean on the choice, or a distinct `finish_reason` value for it.

(1) alone is enough and is the smaller change. Either one lets a caller log the condition and decide for itself; today it can only guess.

**Reproduction:** issue any prompt that makes the model reason past the budget, with `chat_template_kwargs: {"enable_thinking": true}` and `thinking_budget: 1024`, and compare `len(reasoning_content)` against the budget while `finish_reason` reads `stop`. The severing is visible in `reasoning_content`'s last characters on every such call; whether the visible answer also gets cut depends on the model's continuation, which is why the caller needs the signal rather than a rule of thumb.
